cli/tests/secrets_command_test.rs

```diff
@@ -0,0 +1,242 @@
+//! CLI tests for `langstar secrets` commands.
+//!
+//! These tests verify:
+//! - CLI argument parsing and validation
+//! - Help text completeness
+//! - Error handling for invalid inputs
+//! - Security feature validation (no `--value` flag)
+//! - Mutual exclusivity of input method flags
+//!
+//! **Test Categories:**
+//!
+//! 1. **Unit tests** (no API access): Verify CLI parsing, help text, error handling
+//! 2. **Integration tests** (requires API): Verify actual secrets operations
+//!
+//! **Prerequisites for Integration Tests:**
+//!
+//! - `LANGSMITH_API_KEY` environment variable
+//!
+//! Run with: `cargo test --test secrets_command_test`
+
+use assert_cmd::Command;
+use escargot::CargoBuild;
+use predicates::prelude::*;
+
+/// Helper function to get a CLI command builder
+fn langstar_cmd() -> Command {
+    let bin = CargoBuild::new()
+        .bin("langstar")
+        .run()
+        .expect("Failed to build langstar binary")
+        .path()
+        .to_owned();
+    Command::new(bin)
```

Comment:
many cli tests require the binary. does this recompile it every time? and if so, is that expensive or is there a build cache that's reused across all cli integration tests? if it is expensive, can/should we make this a common test fixture?

Reply:
Good question! The `escargot::CargoBuild` helper is designed for this - it caches the built binary and only recompiles when source files change. From the escargot docs: "Escargot is currently designed primarily around building the test binary once and reusing it in tests."\n\nSo the first test that calls `langstar_cmd()` will trigger a build, but subsequent tests in the same run reuse the cached binary. This is efficient and follows the pattern used by other CLI test files in this codebase (model_config_command_test.rs, assistant_command_test.rs, etc.).